### DISCUSSION TITLE
Vehicle VRPTW solution obtain 0 route distanse

### DISCUSSION THREAD

**Discussion by jysun95** [google/or-tools#2515] [Routing (and legacy CP) questions] [answered]
https://github.com/google/or-tools/blob/master/examples/notebook/constraint_solver/vrp_time_windows.ipynb
I've change the time_matrix and time_window with my own data, but I'm facing some problem such as the program only get solution when setting multiple vehicles even if numerous of them indicates 0 route time in the result solution, but if I set vehicle_number to 1, then no solution found, am I missing some parameters?
'
Route for vehicle 0:
0 Time(0,0) -> 0 Time(0,0)
Time of the route: 0min

Route for vehicle 1:
0 Time(0,0) -> 0 Time(0,0)
Time of the route: 0min

Route for vehicle 2:
0 Time(0,0) -> 2 Time(14,60) -> 3 Time(24,70) -> 1 Time(38,84) -> 4 Time(50,96) -> 5 Time(61,107) -> 6 Time(74,120) -> 7 Time(120,145) -> 8 Time(136,161) -> 9 Time(147,172) -> 10 Time(159,184) -> 11 Time(171,196) -> 12 Time(183,208) -> 13 Time(194,219) -> 15 Time(205,230) -> 14 Time(215,240) -> 16 Time(240,360) -> 17 Time(253,468) -> 18 Time(480,480) -> 19 Time(491,500) -> 25 Time(503,512) -> 20 Time(514,523) -> 24 Time(527,536) -> 23 Time(539,548) -> 22 Time(551,560) -> 21 Time(563,572) -> 29 Time(600,600) -> 28 Time(610,610) -> 27 Time(623,623) -> 26 Time(652,652) -> 0 Time(664,664)
Time of the route: 664min

Total time of all routes: 664min'

**Comment by lperron** [comment, marked as answer]
As explained in many places, the first solution heuristic is just a heuristic, and in many case is unable to build the solution found by extensive local search.
